packages/neon_framework/test/dialog_test.dart

```diff
@@ -0,0 +1,171 @@
+
+        await widgetTester.tap(find.byType(ElevatedButton));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+    });
+
+    group('NeonRenameDialog', () {
+      testWidgets('NeonRenameDialog widget', (final widgetTester) async {
+        const title = 'My Title';
+        const value = 'My value';
+        const dialog = NeonRenameDialog(title: title, value: value);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.text(title), findsExactly(2), reason: 'The title is also used for the confirmation button');
+        expect(find.text(value), findsOneWidget);
+        expect(find.byType(TextFormField), findsOneWidget);
+      });
+
+      testWidgets('NeonRenameDialog actions', (final widgetTester) async {
+        const title = 'My Title';
+        const value = 'My value';
+        const dialog = NeonRenameDialog(title: title, value: value);
+
+        // Equal value should not submit
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+        await widgetTester.enterText(find.byType(TextFormField), '');
+        await widgetTester.tap(find.byType(NeonDialogAction));
+        verifyNever<void>(() => mockedObserver.didPop(any(), any()));
+
+        // Different value should submit
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+        await widgetTester.enterText(find.byType(TextFormField), 'My new value');
+        await widgetTester.tap(find.byType(NeonDialogAction));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+    });
+
+    group('NeonErrorDialog', () {
+      testWidgets('NeonErrorDialog widget', (final widgetTester) async {
+        const title = 'My Title';
+        const content = 'My content';
+        var dialog = const NeonErrorDialog(content: content, title: title);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.byIcon(Icons.error), findsOneWidget);
+        expect(find.text(title), findsOneWidget);
+        expect(find.text(content), findsOneWidget);
+        expect(find.byType(NeonDialogAction), findsOneWidget);
+
+        dialog = const NeonErrorDialog(content: content);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.text(NeonLocalizationsEn().errorDialog), findsOneWidget);
+      });
+
+      testWidgets('NeonErrorDialog action', (final widgetTester) async {
+        const content = 'My content';
+        const dialog = NeonErrorDialog(content: content);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        await widgetTester.tap(find.byType(NeonDialogAction));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+    });
+
+    group('NeonDialog', () {
+      testWidgets('NeonDialog', (final widgetTester) async {
+        const dialog = NeonDialog();
+        await widgetTester.pumpWidget(wrapDialog(dialog, TargetPlatform.macOS));
```

Comment:
Why macOS?

Reply:
On cupertino platforms a cancel button is needed as dialogs can not be dismissed by tapping outside of them.
This tests handles this.

I chose macOS to hit both iOS and macOS in the switch of `NeonDialogAction`.

Reply:
I see, then please name the test accordingly